### PR TITLE
Call for PR in yolov5_seg.cpp 

### PR DESCRIPTION
**Pull request by PrinceP**
- [x] Convert class id to the class name
- [x] added confidence
- [x] Different colours for class ids already present
- [x] background box to match with the yolov5 repo
- [x] added coco.txt

seg_zidane

### CODE REVIEW

**Comment by wang-xinyu** [comment]
@PrinceP Thanks for the PR.

To reduce repo size, we don't allow to upload binary images to the repo.
- Can you upload those two images in this thread, and paste the url to the readme?
- For the coco label txt file, can you use this link? https://github.com/amikelive/coco-labels/blob/master/coco-labels-2014_2017.txt and use wget to download https://raw.githubusercontent.com/amikelive/coco-labels/master/coco-labels-2014_2017.txt?

**Comment by PrinceP** [comment]
@wang-xinyu  Done

seg_bus
seg_zidane

File: yolov5/yolov5_seg.cpp [hunk 1]
```diff
@@ -274,7 +275,22 @@ void draw_mask_bbox(cv::Mat& img, std::vector<Yolo::Detection>& dets, std::vecto
 
     cv::rectangle(img, r, bgr, 2);
     // TODO(Call for PR): convert class id to class name
```
Comment (wang-xinyu) [inline yolov5/yolov5_seg.cpp:277]: Can we delete this comment, since it has been addressed.

File: yolov5/yolov5_seg.cpp [hunk 2]
```diff
@@ -217,7 +218,7 @@ std::vector<cv::Mat> process_mask(const float* proto, std::vector<Yolo::Detectio
         e = 1.0f / (1.0f + expf(-e));
         mask_mat.at<float>(y, x) = e;
         // if (e > 0.5) {
-        //   // TODO(Call for PR): Use different colors for different class ids
```
Comment (wang-xinyu) [inline yolov5/yolov5_seg.cpp:220]: Can we delete this comment, since it has been addressed.

File: yolov5/src/utils.h [hunk 3]
```diff
@@ -48,5 +52,46 @@ static inline int read_files_in_dir(const char *p_dir_name, std::vector<std::str
     return 0;
 }
 
+// Function to trim leading and trailing whitespace from a string
+static inline std::string trim_leading_whitespace(const std::string& str)
+{
```
Comment (wang-xinyu) [inline yolov5/src/utils.h:57]: The `{` should not in a new line.

File: yolov5/src/utils.h [hunk 4]
```diff
@@ -48,5 +52,46 @@ static inline int read_files_in_dir(const char *p_dir_name, std::vector<std::str
...
+    size_t first = str.find_first_not_of(' ');
+    if (std::string::npos == first)
+    {
```
Comment (wang-xinyu) [inline yolov5/src/utils.h:60]: The `{` should not in a new line.

File: yolov5/src/utils.h [hunk 5]
```diff
@@ -48,5 +52,46 @@ static inline int read_files_in_dir(const char *p_dir_name, std::vector<std::str
...
+        return str;
+    }
+    size_t last = str.find_last_not_of(' ');
+    return str.substr(first, (last - first + 1));
+}
+// Src: https://stackoverflow.com/questions/16605967
+static inline std::string to_string_with_precision(const float a_value, const int n = 2)
+{
```
Comment (wang-xinyu) [inline yolov5/src/utils.h:68]: The `{` should not in a new line. Can you please check all?

File: yolov5/README.md [hunk 6]
```diff
@@ -128,12 +128,16 @@ wget https://github.com/joannzhang00/ImageNet-dataset-classes-labels/blob/main/i
 # Build and serialize TensorRT engine
 ./yolov5_seg -s yolov5s-seg.wts yolov5s-seg.engine s
 
+# Download the labels file
+wget -O coco.txt https://raw.githubusercontent.com/amikelive/coco-labels/master/coco-labels-2014_2017.txt
+
 # Run inference
-./yolov5_seg -d yolov5s-seg.engine ../samples
+./yolov5_seg -d yolov5s-seg.engine ../samples ../coco.txt
```
Comment (wang-xinyu) [inline yolov5/README.md:135]: Should this be like this?
./yolov5_seg -d yolov5s-seg.engine ../samples coco.txt

**Pull request merged**